examples/gnrc_border_router/Makefile

```diff
@@ -105,19 +80,34 @@ SHOULD_RUN_KCONFIG ?=
 
 include $(RIOTBASE)/Makefile.include
 
-ifndef CONFIG_GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE
-ifeq (1,$(STATIC_ROUTES))
-  CFLAGS += -DCONFIG_GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE=1
-  # CONFIG_GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE=1 requires one more address for
-  # `fe80::2`.
-  CFLAGS += -DCONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF=3
-endif
+# Compile-time configuration for DHCPv6 client (needs to come after
+# Makefile.include as this might come from Kconfig)
+ifeq (1,$(USE_DHCPV6))
+  ifndef CONFIG_GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE
+    ifeq (1,$(STATIC_ROUTES))
+      CFLAGS += -DCONFIG_GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE=1
+      # CONFIG_GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE=1 requires one more address
+      # for `fe80::2`.
+      CFLAGS += -DCONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF=3
+    endif
+  endif
 endif
 
+
 .PHONY: host-tools
 
 host-tools:
 	$(Q)env -u CC -u CFLAGS make -C $(RIOTTOOLS)
 
+# define native specific targets to only run UHCP daemon when required
+ifneq (,$(filter native,$(BOARD)))
+ifneq (1,$(USE_DHCPV6))
+.PHONY: uhcpd-daemon
+
+uhcpd-daemon: host-tools
+	$(RIOTTOOLS)/uhcpd/bin/uhcpd $(TAP) $(IPV6_PREFIX) &
+endif
+endif
```

Comment:
Might as well move this to `Makefile.native.conf` then.

Reply:
Since `host-tools` is defined here, I thought as this target is dependent on it (and more about `uhcpd` than `native` - other boards if they use e.g. Ethernet might also want to use it) to rather put it in the main `Makefile`.